gizmo.py

```diff
@@ -0,0 +1,104 @@
+
+import glob
+import os
+import warnings
+import numpy as np
+import pandas as pd
+
+def hello(name=None,country='Finland'):
+    if name is not None:
+        out_text = 'Hello {name}, how are things in {country}?'\
+            .format(name=name,country=country)
+        print(out_text)
+    else:
+        print('you need to enter a valid name')
+        
+def spell(string=''):
+    output = ''
+    for char in string:
+        output=output + '.'+ char
+    output = output[1:]
+    print(output)
+
+def relative_path(identifier_list):
+    output = []
+    if isinstance(identifier_list, str) or \
+    isinstance(identifier_list, int) or \
+    isinstance(identifier_list, float):
+        return('./subjects/mock_recording_{subject_identifier}.rec')\
+                        .format(subject_identifier = str(identifier_list))
+    
+    if hasattr(identifier_list,'__iter__'):
+        if len(identifier_list) != 0:
+            for file in identifier_list:
+                if isinstance(file,str):
+                    output.append('./subjects/mock_recording_{subject_identifier}.rec'\
+                        .format(subject_identifier = str(file)))
+                else:
+                    print('object in the list is not a string!')
+                    
+        else:
+            warnings.warn('The supplied itterator is empty')
+            
+    else:
+        warnings.warn('Supplied object is not an itterator')
+    return(output)
+
+class Gizmo:
+    def __init__(self,name):
+        self.name = name
+    
+    def speak(self):
+        print(self.name)
+    
+def multiplication_table(zero_out_multiples: int =0, table_size: int=12) -> np.ndarray:
+    """Returns the multiplitation table of a number
+    
+    This function returns the multiplitation table of a number
+    with ommited values for specific integer that values in table 
+    are its multiplier.
+    
+    Parameters
+    ----------
+        zero_out_multiples: int (optional)
+            This vale replaces its mulipliers with 0.
+            if multiplier is not specified, nothing will change.
+            multiplier with value of zero is discarded.
+            
+        table_size: int 
+            maximum size of the table.
+            
+    Returns
+    -------
+        np.ndarray
+            multiplication table with removed values of 
+            `zero_out_multiples` if specified
+    """
+    number = table_size
+    array_ = np.array([list(range(number))])+1
```

Comment:
A lot going on there.. you make a range, then convert it to a list, pack it inside yet another list, then convert it to an array. Are you aware of the `np.arange` function?

```python
array_ = np.arange(number) + 1
```